### PR TITLE
chore(Drawer): rely on FocusTrap returnFocus for focus restoration

### LINKED ISSUE #1589: Unify focus-restoration logic in FocusTrap and its consumers
## Context

Follow-up from [PR #1578](https://github.com/Telefonica/mistica-web/pull/1578) ([review comment](https://github.com/Telefonica/mistica-web/pull/1578/files#r3469151486)).

That PR changed `FocusTrap`'s default to `returnFocus = true` so the trigger element regains focus when the trap unmounts. However, some consumers already restore focus manually, so there are now two competing mechanisms.

## Problem

`FocusTrap` is consumed by `Drawer`, `Dialog`, `Sheet`, and `NavigationBar`. With `returnFocus = true` as the default, focus restoration can happen twice in consumers that also restore focus themselves:

- `Drawer` has its own `useRestoreFocus` hook (`src/drawer.tsx`) that focuses the previously active element on unmount.

Duplicate restoration can lead to redundant `focus()` calls and potentially unexpected scroll/focus events.

## Proposal

Establish a single source of truth for focus restoration:

- Either rely on `FocusTrap`'s `returnFocus` everywhere and remove the consumers' bespoke restore logic (e.g. `Drawer`'s `useRestoreFocus`), or
- Keep the bespoke logic where it is needed and disable `returnFocus` in those consumers.

Audit all `FocusTrap` consumers (`drawer.tsx`, `dialog.tsx`, `sheet-common.tsx`, `navigation-bar.tsx`) and converge on one consistent approach.

## Notes

This was intentionally deferred from PR #1578 to keep that change focused; it deserves a proper analysis on its own.

### PR DESCRIPTION
**Pull request by brtbrt**
## Context

Follow-up from PR #1578, which changed `FocusTrap`'s default to `returnFocus = true`. With that default, `Drawer` restored focus twice on close: once through `FocusTrap` (`react-focus-lock`'s `returnFocus`) and once through its own `useRestoreFocus` hook. This produced redundant `focus()` calls.

## Changes

- Remove `Drawer`'s `useRestoreFocus` hook and rely on `FocusTrap`'s `returnFocus` as the single source of truth, matching `Dialog` and `Sheet`.
- Add focus-restoration tests for `Drawer` and `Sheet`, which previously had none. `Dialog` was already covered.
- Remove the resolved TODO in `FocusTrap`.

## Audit notes

All four `FocusTrap` consumers were reviewed. `Dialog`, `Sheet`, and `NavigationBar` already relied on `FocusTrap`; only `Drawer` carried bespoke logic. `NavigationBar` is intentionally excluded: its `FocusTrap` wraps the bar including the burger trigger, so focus is retained on the trigger automatically and `returnFocus` performs no restoration there, leaving nothing meaningful to test.

Closes #1589

### CODE REVIEW

File: src/__tests__/drawer-test.tsx [hunk 1]
```diff
@@ -63,6 +63,24 @@ test.each(['esc', 'overlay', 'x'])('Drawer dismiss: %s', async (dismissMethod: s
     });
 });
 
+test('Drawer restores focus to the trigger when closed', async () => {
+    render(<DrawerTest />);
+
+    const openButton = screen.getByRole('button', {name: 'open'});
+    await userEvent.click(openButton);
+
```
[the remark this answers is not in the document]
Comment (brtbrt) [reply src/__tests__/drawer-test.tsx:72]: u r right. fixed.

File: AGENTS.md [hunk 2]
```diff
@@ -28,3 +28,34 @@ React component library for Telefonica's Design System.
 
 - Read [CONTRIBUTING](./CONTRIBUTING.md) before creating a PR or contributing to Mistica repo.
 - Always add the `AI` label to PRs where the code was written by an AI agent.
+
```
Comment (brtbrt) [inline AGENTS.md]: not sure I wanna introduce it here. This is an attempt to check if AI better performs with a context about the documentation it can exploit, without any further ls commands, and I/O of tokens.

That said, I've created a separated branch to perform an A/B test on it

https://github.com/Telefonica/mistica-web/pull/1593/changes

File: doc/commits-and-prs.md
Comment (brtbrt) [inline doc/commits-and-prs.md:1]: I've experienced problems with opus' commits: it used to use the whole semantic versioning's types, but we're using a subset. with this doc, we instruct it about docs, PRs etc, if it needs to commit/push/creating PR etc

**Comment by brtbrt** [comment]
a11y report that `http://localhost:6006/iframe.html?viewMode=story&id=components-accordions--accordion-story&skin=O2-new` fails, but locally passes. do we have any known problem with CI?

File: .github/actions/audit-accessibility/main.js [hunk 3]
```diff
@@ -73,11 +73,26 @@ const audit = async (browser, url, disabledRules = []) => {
     const page = await browser.newPage();
     const ua = await browser.userAgent();
     await page.setUserAgent(`${ua} acceptance-test`);
-    await page.goto(url);
+    await page.goto(url, {waitUntil: 'networkidle0'});
```
Comment (brtbrt) [inline .github/actions/audit-accessibility/main.js:76]: I've had problems with the CI executing a11y axe scan. apparently, axe was scanning the page that it was still rendering (no html tag found). With this fix it looks stable, without timing issues

File: doc/commits-and-prs.md
Comment (Marcosld) [reply doc/commits-and-prs.md:1]: Commits don't matter as we are squashing into main. I'd remove this file altogether to not fill up llm context. We can move pieces we really find relevant and not already stated into CONTRIBUTING.md. I've also had problems but they are cause by claude not reading CONTRIBUTING.md. We can try modifying agents to check if it improves reading probability.

File: doc/commits-and-prs.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+type subset deliberately limits which increments occur automatically:
+
+- `feat` raises the **minor** version.
+- `fix` and `revert` raise the **patch** version.
+- `chore` produces **no release**.
+
+Major-version increments are not produced through routine pull requests. Breaking changes are
+coordinated and released deliberately rather than triggered automatically from a commit type.
+
+## Pull-request rules
```
Comment (Marcosld) [inline doc/commits-and-prs.md]: All this is already stated in contributing.md

File: doc/commits-and-prs.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+- The description should be a concise summary of the problem and the fix, ending with
+  `Ref: <TICKET-ID>` when a Jira ticket exists.
+- Add the `@Telefonica/mistica-web-reviewers` team as a reviewer.
+- Add the `AI` label when the code was written by an AI agent.
+
+Only the pull-request title is validated. Local commit messages are not linted; the `pre-commit`
+hook runs `lint-staged` only. This works because pull requests are squashed on merge, so the title
+is the single message that reaches `master`.
+
+## Release pipeline
```
Comment (Marcosld) [inline doc/commits-and-prs.md]: why?

File: doc/commits-and-prs.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+Conventional Commits defines many types, but this repository intentionally supports only a subset.
+The allowlist is enforced by the `validate-pr-title` job in
+[`.github/workflows/ci.yml`](../.github/workflows/ci.yml):
+
+| Type     | Purpose                                          | Release effect   |
+| -------- | ------------------------------------------------ | ---------------- |
+| `feat`   | A new feature or a new component API.            | Minor version    |
+| `fix`    | A bug fix.                                        | Patch version    |
+| `revert` | Reverts a previous change.                        | Patch version    |
+| `chore`  | Internal work with no user-facing impact.         | No release       |
```
Comment (Marcosld) [inline doc/commits-and-prs.md]: Perhaps we can try adding these types to CONTRIBUTING.md instead of creating a new file.

File: AGENTS.md [hunk 7]
```diff
@@ -28,3 +28,34 @@ React component library for Telefonica's Design System.
 
 - Read [CONTRIBUTING](./CONTRIBUTING.md) before creating a PR or contributing to Mistica repo.
 - Always add the `AI` label to PRs where the code was written by an AI agent.
```
Comment (Marcosld) [inline AGENTS.md]: I think what we have to improve is this part... it sometimes skips reading the file. If we can't manage to make it read it we will have to move relevant info here.

File: AGENTS.md (hunk 2 above)
Comment (Marcosld) [reply AGENTS.md]: Ok, better test it in another branch. Those files are meant for mistica consumers, not mistica contributors, but perhaps they can help in some way.

File: .github/actions/audit-accessibility/main.js [hunk 8]
```diff
@@ -73,11 +73,26 @@ const audit = async (browser, url, disabledRules = []) => {
...
+        )
+        .catch(() => {
+            // Some stories may legitimately render no content; continue with the audit.
+        });
     const result = await new AxePuppeteer(page)
         .disableRules([
             // ignored because some stories don't include an H1 header
             'page-has-heading-one',
+            // ignored because isolated component stories have no <main> landmark
+            'landmark-one-main',
```
Comment (Marcosld) [inline .github/actions/audit-accessibility/main.js]: Are you sure? Axe was working fine without disabling this rule

**Review by Marcosld — CHANGES_REQUESTED** [review]

**Comment by AlexandraGallipoliRodrigues** [comment]
should the PR title be `fix(Drawer): ...` instead of `chore(Drawer): ...`?

File: doc/commits-and-prs.md (hunk 4 above)
Comment (brtbrt) [reply doc/commits-and-prs.md]: yep let's merge what's useful here into CONTRIBUTING

File: doc/commits-and-prs.md (hunk 6 above)
Comment (brtbrt) [reply doc/commits-and-prs.md]: agreed!

File: AGENTS.md (hunk 2 above)
Comment (brtbrt) [reply AGENTS.md]: yeah, that's a really good point: we don't have doc related to the repo itself: we have doc focused on the library's consumers. I'll try to reorganize the thing on the other branch, so that we can take advantage of the two worlds

File: AGENTS.md (hunk 7 above)
Comment (brtbrt) [reply AGENTS.md]: github convention is way too misleading iMHO. But let's move trials and stuff on the dedicated branch. meanwhile we'll get rid of the documentation section here below

File: .github/actions/audit-accessibility/main.js (hunk 8 above)
Comment (brtbrt) [reply .github/actions/audit-accessibility/main.js]: I'm sure I'm never sure! I've had problems with that, but let's try one more time without it. less is more

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -21,12 +21,31 @@ Don't hesitate to ask any questions and share your ideas
...
+
+#### Allowed types (the supported subset)
+
+Conventional Commits defines many types, but this repository intentionally supports only a subset.
+The allowlist is enforced by the `validate-pr-title` job in
+[`.github/workflows/ci.yml`](../.github/workflows/ci.yml#116):
+
+### PR Description
+Concise summary of the problem and fix, ending with `Ref: <ISSUE-ID>`;
+
```
[the remark this answers is not in the document]
Comment (brtbrt) [reply CONTRIBUTING.md:49]: period is used when a list ends. semicolon is the right standard to further adds other points below

**Comment by brtbrt** [comment]
@AlexandraGallipoliRodrigues  not really: this is not fixing up anything really, we're simply removing a duplicated behavior we have  on a component :). but it's not causing any misbehavior

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -21,12 +21,31 @@ Don't hesitate to ask any questions and share your ideas
...
+  changes that do not touch a component, use `Chore` as the scope.
+- **subject**: a concise, imperative description of the change.
+- Add the `AI` label when the code was written by an AI agent.
+
+#### Allowed types (the supported subset)
+
+Conventional Commits defines many types, but this repository intentionally supports only a subset.
+The allowlist is enforced by the `validate-pr-title` job in
+[`.github/workflows/ci.yml`](../.github/workflows/ci.yml#116):
+
```
[the remark this answers is not in the document]
Comment (brtbrt) [reply CONTRIBUTING.md:42]: is it worth? maybe because we spare some I/O round? but it could be get stale easily

File: AGENTS.md [hunk 11]
```diff
@@ -24,7 +24,7 @@ React component library for Telefonica's Design System.
 
 ---
 
-## GitHub conventions
+## Conventions
```
Comment (Marcosld) [inline AGENTS.md:27]: why rename?

File: CONTRIBUTING.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (Marcosld) [reply CONTRIBUTING.md:42]: Hmmmm perhaps it is worth to add the allowlist, it doesn't usually change...

File: CONTRIBUTING.md
Comment (Marcosld) [inline CONTRIBUTING.md:1]: I'd leave the file as it was, we aren't really adding any info with these changes.

**Review by Marcosld — APPROVED** [review]

File: AGENTS.md (hunk 11 above)
Comment (brtbrt) [reply AGENTS.md:27]: because "github" can be misleading for the AI
